src/platform_linux/shmem.c

```diff
@@ -169,12 +169,13 @@ platform_shm_heap_valid(shmem_heap *shmheap);
  * the heap-ID handle to the shared memory's start address, from which the
  * location of the next-free-byte can be tracked.
  */
-static inline shmem_heap *
+shmem_heap *
 platform_heap_id_to_shmaddr(platform_heap_id hid)
 {
    debug_assert(hid != NULL);
-   void *shmaddr = ((void *)hid - offsetof(shmem_heap, shm_id));
-   return (shmem_heap *)shmaddr;
+   shmem_heap *shmaddr =
```

Comment:
I don't understand why you don't just make the hid for a shared-mem heap to be equal to the address of its control block.

In other words, why isn't this code just
```
   return (shmem_heap *)hid;
```

Reply:
OK; Done. This needed a small fix in `platform_shmcreate()`, to return the heap_id correctly when splinter is first created. That new code now looks like:

```
 331 platform_status
 332 platform_shmcreate(size_t            size,
 333                    platform_heap_id *heap_id) // Out
[...]
 377    // Return 'heap-ID' handle pointing to start addr of shared segment.
 378    if (heap_id) {
 379       *heap_id = (platform_heap_id *)shmaddr;
 380    }
```

Unit-tests seem to pass with this new approach. Will watch CI ...